### PR TITLE
Introduction to the KeyPair.

### PR DESCRIPTION
**Pull request by nothingalike**

### CODE REVIEW

**Comment by nothingalike** [comment]
Ok so the main things here are:
 - Added `KeyPair` this is primarily for generating a random key pair. Useful for one off key pairs for Minting and MultiSig Policies. Stake pool support also comes to mine.
 - I added `Sign` and `Verify` to the `PrivateKey` and `PublicKey` and consolidated all logic around this.

**Comment by nothingalike** [comment]
Also I'm open to suggestions for more testing around Sign and Verify

File: CardanoSharp.Wallet.Test/TransactionTests.cs [hunk 1]
```diff
@@ -304,8 +304,8 @@ public void MnemonicToTransaction()
                 .WithVKeyWitnesses(new List<VKeyWitness>()
                 {
                     new VKeyWitnessBuilder()
-                        .WithVKey(paymentPub.Key)
-                        .WithSKey(paymentPrv.Key)
+                        .WithVKey(paymentPub)
+                        .WithSKey(paymentPrv)
                         .Build()
```
Comment (Daniel-SchaeferJ) [inline CardanoSharp.Wallet.Test/TransactionTests.cs:309]: You know I was just thinking about this VKeyWitnessBuilder and how FREAKIN easy it'll be to make a pool manager with this library. It makes me think though that we do need a cddl spec converter to make a pool manager easy for us

Comment (nothingalike) [reply CardanoSharp.Wallet.Test/TransactionTests.cs:309]: I'm always up for a cddl converter. I just don't want to be the one to code that up lol

**Review by Daniel-SchaeferJ — APPROVED** [review]
This looks good. I really like the direction the library is taking on the front of being modular and the types coming from this. I love the fact the days of our code littered with byte[] are coming to an end so overall good job.

File: CardanoSharp.Wallet.Test/TransactionTests.cs (hunk 1 above)
Comment (Daniel-SchaeferJ) [reply CardanoSharp.Wallet.Test/TransactionTests.cs:309]: Yeah me neither, and I don't think we will need it until we want to look into the prospect of a pool manager.

**Pull request merged**